packages/pymedphys_dicom/src/pymedphys_dicom/dicom/dose.py

```diff
@@ -32,100 +30,148 @@
 import matplotlib.pyplot as plt
 from matplotlib import path
 
+from scipy.interpolate import RegularGridInterpolator
+
 import pydicom
 import pydicom.uid
 
-from .structure import pull_structure
-from .coords import coords_from_xyz_axes, xyz_axes_from_dataset
+from ..rtplan import get_surface_entry_point
 
+from .structure import pull_structure
+from .coords import xyz_axes_from_dataset
 
 # pylint: disable=C0103
 
 
 def zyx_and_dose_from_dataset(dataset):
     x, y, z = xyz_axes_from_dataset(dataset)
     coords = (z, y, x)
-    dose = dose_from_dataset(dataset, reshape=False)
+    dose = dose_from_dataset(dataset)
 
     return coords, dose
 
 
-def dose_from_dataset(ds, set_transfer_syntax_uid=True, reshape=True):
+def dose_from_dataset(ds, set_transfer_syntax_uid=True, reshape=False):
     r"""Extract the dose grid from a DICOM RT Dose file.
-
-    .. deprecated:: 0.5.0
-            `dose_from_dataset` will be removed in a future version of
-            PyMedPhys in favour of a new & improved API.
     """
 
     if set_transfer_syntax_uid:
         ds.file_meta.TransferSyntaxUID = pydicom.uid.ImplicitVRLittleEndian
 
     if reshape:
-        warnings.warn((
-            '`dose_from_dataset` currently reshapes the dose grid. In a '
-            'future version this will no longer occur. To begin using this '
-            'function without the reshape pass the parameter `reshape=False` '
-            'when calling `dose_from_dataset`.'), UserWarning)
-        pixels = np.transpose(
-            ds.pixel_array, (1, 2, 0))
-    else:
-        pixels = ds.pixel_array
+        warnings.warn(
+            ('The `reshape` parameter no longer does anything. Please remove '
+             'this parameter. In a future version this parameter will no '
+             'longer be accepted.'), UserWarning)
 
-    dose = pixels * ds.DoseGridScaling
+    dose = ds.pixel_array * ds.DoseGridScaling
 
     return dose
 
 
-def axes_and_dose_from_dicom(dicom_filepath):
-    ds = pydicom.dcmread(dicom_filepath, force=True)
-    axes = xyz_axes_from_dataset(ds)
-    dose = dose_from_dataset(ds)
+def dicom_dose_interpolate(interp_coords, dose: pydicom.Dataset):
+    """Interpolates across a DICOM dose dataset.
+
+    Parameters
+    ----------
+    dose : pydicom.Dataset
+        An RT DICOM Dose object
+    interp_coords : tuple(z, y, x)
+        A tuple of coordinates in DICOM order, z axis first, then y, then x
```

Comment:
Oh? This distinction hadn't hit home with me. Honestly, I'd like us to pick one convention and stick with it everywhere...

Reply:
Fair enough, haha!

I tried to explain it [here](https://github.com/pymedphys/pymedphys/blob/c0484461e8ee731c8887798d2066086fd49269a8/packages/pymedphys_dicom/src/pymedphys_dicom/dicom/coords.py#L31)

Reply:
Yeah, the more I look at `xyz_axes_from_dataset()` and `coords_from_xyz_axes()`, I think axis/axes are a bad choice.

The former function returns a tuple of length == 3, with the first tuple element containing a 1-D array containing the full range of `x` values that appear as coordinates for the dose grid (and so on for `y` and `z`).

I tried to avoid the word "coordinates", because I wanted to reserve that word for the tuples that contain meshgrids, such that `coords[a, b, c]` corresponds to the coordinate triplet for `dosegrid[a, b, c].

So, I went for "axes". But I'm thinking that's also confusing. What do you think about "ticks"?

Reply:
Hmm, not convinced here really. I think this is a place where the concept of a well defined object that makes it hard to mess up the coordinate space would really help.

Reply:
See https://github.com/pymedphys/pymedphys/issues/431 :)

Reply:
To be resolved within https://github.com/pymedphys/pymedphys/issues/431